test/reshape.jl

```diff
@@ -347,5 +347,20 @@ end
                     renamecols=x->string("_", x, "_"))
     @test wide3 == w2[!, Not(2)]
 end
+#
+# flatten(df::AbstractDataFrame, veccol::Union{Integer, Symbol})
+#
+@testset "flatten" begin
+    df_vec = DataFrame(a = [1, 2], b = [[1, 2], [3, 4]])
+    df_tup = DataFrame(a = [1, 2], b = [(1, 2), (3, 4)])
+    ref = DataFrame(a = [1, 1, 2, 2], b = [1, 2, 3, 4])
+    @test flatten(df_vec, :b) == flatten(df_tup, :b) == ref
+    df = DataFrame(a = [1, 2], b = [[1, 2], ["x", "y"]])
+    ref = DataFrame(a = [1, 1, 2, 2], b = [1, 2, "x", "y"])
+    @test flatten(df, :b) == ref
+    df = DataFrame(a = [1, 2], b = [(i for i in 1:5), (i for i in 6:10)])
+    ref = DataFrame(a = [fill(1, 5); fill(2, 5)], b = collect(1:10))
+    @test flatten(df, :b) == ref
+end
```

Comment:
You mentioned a failure with CategoricalArrays. Can you add a test for that? That applies both to columns to flatten and to other columns.

It would also be good to have a test where vectors to flatten have different element types. One particularly tricky case to check is e.g. `Union{Missing, Int}[1, 2]}` the `Missing` type should be preserved.